app/src/main/cpp/skyline/gpu/texture/layout.cpp

```diff
@@ -89,144 +96,339 @@ namespace skyline::gpu::texture {
     }
 
     /**
-     * @brief Copies pixel data between a linear and blocklinear texture
-     * @tparam BlockLinearToLinear Whether to copy from a blocklinear texture to a linear texture or a linear texture to a blocklinear texture
+     * @brief Copies pixel data between a pitch-linear and blocklinear texture
+     * @tparam BlockLinearToPitch Whether to copy from a blocklinear texture to a pitch-linear texture or a pitch-linear texture to a blocklinear texture
      */
-    template<bool BlockLinearToLinear>
+    template<bool BlockLinearToPitch>
     void CopyBlockLinearInternal(Dimensions dimensions,
-                                 size_t formatBlockWidth, size_t formatBlockHeight, size_t formatBpb,
+                                 size_t formatBlockWidth, size_t formatBlockHeight, size_t formatBpb, u32 pitchAmount,
                                  size_t gobBlockHeight, size_t gobBlockDepth,
-                                 u8 *blockLinear, u8 *linear) {
+                                 u8 *blockLinear, u8 *pitch) {
         size_t robWidthUnalignedBytes{util::DivideCeil<size_t>(dimensions.width, formatBlockWidth) * formatBpb};
         size_t robWidthBytes{util::AlignUp(robWidthUnalignedBytes, GobWidth)};
         size_t robWidthBlocks{robWidthUnalignedBytes / GobWidth};
 
+        if (formatBpb == 12) [[unlikely]]
+            formatBpb = 4;
```

Comment:
Do you not need to change the width or something here?

Reply:
No, because the width is only used in bytes, which does not change